app/models/miq_server/worker_management/heartbeat.rb

```diff
@@ -91,11 +91,23 @@ def validate_heartbeat(w)
     end
   end
 
+  def clean_heartbeat_files
+    Dir.glob(Rails.root.join("tmp", "*.hb")).each { |f| File.delete(f) }
+  end
+
   private
 
-  def workers_last_heartbeat(pid)
+  def workers_last_heartbeat(w)
+    ENV["WORKER_HEARTBEAT_METHOD"] == "file" ? workers_last_heartbeat_to_file(w) : workers_last_heartbeat_to_drb(w)
```

Comment:
I thought of that too (I'm telling the truth!) and discussed it with @jrafanie and we thought that consistency in method names would be better than having them semantically correct.